### ISSUE TITLE
Sequential fit accuracy does not increase with higher epochs

### ISSUE THREAD

**Issue by wooo0214** [wooo0214/MyTools#2]
`model3=Sequential()
model3.add(Conv2D(filters=6, kernel_size=(5,5), strides=(1,1), activation='tanh', input_shape=(256,256,3)))
model3.add(AveragePooling2D(pool_size=(2,2), strides=(2,2)))
model3.add(Conv2D(filters=16, kernel_size=(5,5), strides=(1,1), activation='tanh'))
model3.add(AveragePooling2D(pool_size=(2,2), strides=(2,2)))
model3.add(Flatten())
model3.add(Dense(units=120, activation='tanh'))
model3.add(Dense(units=84, activation='tanh'))
model3.add(Dense(units=10, activation='softmax'))
model_optimizer = Adam(learning_rate=0.001)

reduceLR = ReduceLROnPlateau(monitor='accuracy', factor=.001, patience=1, min_delta=0.01, mode="auto")

model3.compile(optimizer=model_optimizer, loss='sparse_categorical_crossentropy', metrics=["accuracy"])

model3.fit(x_train, y_train, epochs=10, callbacks=[reduceLR]) `

Output:
`Epoch 1/10
[1m444/444[0m [32m━━━━━━━━━━━━━━━━━━━━[0m[37m[0m [1m90s[0m 186ms/step - accuracy: 0.1341 - loss: 2.3357 - learning_rate: 0.0010
Epoch 2/10
[1m444/444[0m [32m━━━━━━━━━━━━━━━━━━━━[0m[37m[0m [1m84s[0m 189ms/step - accuracy: 0.1361 - loss: 2.2382 - learning_rate: 0.0010
Epoch 3/10
[1m444/444[0m [32m━━━━━━━━━━━━━━━━━━━━[0m[37m[0m [1m77s[0m 173ms/step - accuracy: 0.1448 - loss: 2.2475 - learning_rate: 1.0000e-06
Epoch 4/10
[1m444/444[0m [32m━━━━━━━━━━━━━━━━━━━━[0m[37m[0m [1m89s[0m 200ms/step - accuracy: 0.1481 - loss: 2.2476 - learning_rate: 1.0000e-09
Epoch 5/10
[1m444/444[0m [32m━━━━━━━━━━━━━━━━━━━━[0m[37m[0m [1m88s[0m 198ms/step - accuracy: 0.1468 - loss: 2.2438 - learning_rate: 1.0000e-12
Epoch 6/10
[1m444/444[0m [32m━━━━━━━━━━━━━━━━━━━━[0m[37m[0m [1m88s[0m 199ms/step - accuracy: 0.1491 - loss: 2.2451 - learning_rate: 1.0000e-15
Epoch 7/10
[1m444/444[0m [32m━━━━━━━━━━━━━━━━━━━━[0m[37m[0m [1m95s[0m 213ms/step - accuracy: 0.1505 - loss: 2.2388 - learning_rate: 1.0000e-18
Epoch 8/10
[1m444/444[0m [32m━━━━━━━━━━━━━━━━━━━━[0m[37m[0m [1m119s[0m 161ms/step - accuracy: 0.1462 - loss: 2.2452 - learning_rate: 1.0000e-21
Epoch 9/10
[1m444/444[0m [32m━━━━━━━━━━━━━━━━━━━━[0m[37m[0m [1m71s[0m 160ms/step - accuracy: 0.1479 - loss: 2.2435 - learning_rate: 1.0000e-24
Epoch 10/10
[1m444/444[0m [32m━━━━━━━━━━━━━━━━━━━━[0m[37m[0m [1m69s[0m 154ms/step - accuracy: 0.1396 - loss: 2.2460 - learning_rate: 1.0000e-27
`

**Comment by wooo0214** [comment]
1. What does learning_rate do in Sequential and how does it affect the accuracy?
2. What does the loss "sparse_categorical_crossentropy" do here? How do different loss function affect? 
3. What is the reduceLR?

**Comment by wooo0214** [comment]
1st try -- kernel size and strides :
- kernel_size set to (20,20), strides set to (4,4) for 2 convolution layers
- epochs set to 5
>>> accuracy: 0.1724, 0.2214, 0.2373 ,0.2136, 0.2179 
Accuracy increased from 1st ~ 3rd epoch, decreased since the 4th. Better than the original kernel sizes and strides.

https://stackoverflow.com/questions/45538980/how-does-the-size-of-the-patch-kernel-impact-the-result-of-a-convnet
large kernel size may loss details, while a small kernel may lead to overfitting.

In model3 the accuracy grows slowly from epoch 1~4 and then stagnate. In model4 it stagnate from epoch 4. Possibly they are both overfitted.

**Comment by wooo0214** [comment]
2nd try -- kernel size:
- kernel_size set to (9,9); strides remain as (4,4) for 2 convolution layers

> 1. Accuracy doubled and increasing: 0.2, 0.3, 0.4 , 0.5202, 0.5643 
> 2. Faster! It took 70+s per epoch in model 3&4; now it took only 7s per epoch (15s in 1st epoch).

**Comment by wooo0214** [comment]
Closing this issue as the model was overfitting previously, and that's why the accuracy stagnates. 

Splitting "Optimizing accuracy" as a new issue.
